ably/realtime/connection.py

```diff
@@ -159,7 +160,12 @@ async def connect(self):
             if self.__connected_future is None:
                 log.fatal('Connection state is CONNECTING but connected_future does not exist')
                 return
-            await self.__connected_future
+            try:
+                await self.__connected_future
+            except asyncio.CancelledError:
+                exception = AblyException("Connection cancelled due to request timeout", 504, 50003)
```

Comment:
I think we can remove this timeout if we already have one in `connect_impl`

Reply:
The future here raises a cancelled error because it has was cancelled in `connect_impl`. This is to catch the error, for some reason `.cancelled()` didn't work.

Reply:
ahh i see, thanks for explaining! i wonder if it would be better to just do `self.__connected_future.set_exception()` as soon as you catch the `TimeoutError`?

Reply:
ahh i see, thanks for explaining! i wonder if it would be better to just do `self.__connected_future.set_exception()` as soon as you catch the `TimeoutError`?

Reply:
also i notice that you don't do this at all for calling `attach` while the channel state is `ATTACHING` (and the equivalent for `ping` and `detach`) so we should update those implementations too